src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1623,27 +1625,14 @@ public Status do_GET_Document(Database _db, String docID, String _attachmentName
                 String revID = getQuery("rev");  // often null
                 RevisionInternal rev = null;
                 if (isLocalDoc) {
+                    // query only -> not required synchronized
                     rev = db.getLocalDocument(docID, revID);
                 } else {
-                    /*
-                    rev = db.getDocument(docID, revID, options);
-                    // Handle ?atts_since query by stubbing out older attachments:
-                    //?atts_since parameter - value is a (URL-encoded) JSON array of one or more revision IDs.
-                    // The response will include the content of only those attachments that changed since the given revision(s).
-                    //(You can ask for this either in the default JSON or as multipart/related, as previously described.)
-                    List<String> attsSince = (List<String>) getJSONQuery("atts_since");
-                    if (attsSince != null) {
-                        String ancestorId = db.findCommonAncestorOf(rev, attsSince);
-                        if (ancestorId != null) {
-                            int generation = RevisionInternal.generationFromRevID(ancestorId);
-                            db.stubOutAttachmentsIn(rev, generation + 1);
-                        }
-                    }
-                    */
                     boolean includeAttachments = options.contains(TDContentOptions.TDIncludeAttachments);
                     if (includeAttachments) {
                         options.remove(TDContentOptions.TDIncludeAttachments);
                     }
+                    // query only -> not required synchronized
```

Comment:
As [getDocument()](https://github.com/couchbase/couchbase-lite-java-core/blob/a35361bc47ecb6c55849694deece22fa4aaf7c62/src/main/java/com/couchbase/lite/store/SQLiteStore.java#L691) does not access shard instance and SQLiteDatabae is thread-safe as default, I believe this method is thread-safe.